packages/xo-web/src/common/select-boot-firmware.js

```diff
@@ -1,48 +1,62 @@
 import _ from 'intl'
+import decorate from 'apply-decorators'
 import PropTypes from 'prop-types'
 import React from 'react'
 import { confirm } from 'modal'
+import { createGetObject } from 'selectors'
 import { injectState, provideState } from 'reaclette'
-import { noop } from 'utils'
+import { satisfies as versionSatisfies } from 'semver'
+
+import { connectStore, noop } from './utils'
 
-// https://docs.citrix.com/en-us/citrix-hypervisor/whats-new/experimental.html
 // XAPI values should be lowercased
 const VM_BOOT_FIRMWARES = ['bios', 'uefi']
 
-const withState = provideState({
-  effects: {
-    handleBootFirmwareChange(__, { target: { value } }) {
-      if (value !== '') {
-        // TODO: Confirm should be removed once the feature is stabilized
-        confirm({
-          title: _('vmBootFirmware'),
-          body: _('vmBootFirmwareWarningMessage'),
-        }).then(() => this.props.onChange(value), noop)
-      } else {
-        this.props.onChange(value)
-      }
+const SelectBootFirmware = decorate([
+  connectStore({
+    host: createGetObject((_, props) => props.host),
+  }),
+  provideState({
+    effects: {
+      handleBootFirmwareChange(__, { target: { value } }) {
+        if (
+          value !== '' &&
+          this.props.host !== undefined &&
```

Comment:
Should we display the warning in case of an `undefined` host?

Reply:
No, the warning is related to the host version.